src/utilities/identities/identities.ts

```diff
@@ -244,59 +198,66 @@ export async function getIdentityCryptoFromSeed(
     ? deriveEncryptionKeyLegacy(seed)
     : deriveEncryptionKeyFromSeed(seed);
 
-  const { address } = getKeypairBySeed(seed);
+  const keypair = Crypto.makeKeypairFromSeed(seed, 'sr25519');
   const identities = await getIdentities();
-  const did = getIdentityDid(identities[address]);
+  const did = getIdentityDid(identities[keypair.address]);
 
-  const didDetails = await getDidDetails(did);
-  const keystore = await getKeystoreFromSeed(seed);
+  const didDocument = await getDidDocument(did);
 
-  const encryptionKeystore: Pick<NaclBoxCapable, 'encrypt'> = {
-    async encrypt({ data, alg, peerPublicKey }) {
-      const { sealed, nonce } = naclSeal(
-        data,
-        encryptionKey.secretKey,
-        peerPublicKey,
-      );
-
-      return {
-        data: sealed,
-        alg,
-        nonce,
-      };
-    },
-  };
+  async function encryptCallback({ data, peerPublicKey }: EncryptRequestData) {
+    const { box, nonce } = Crypto.encryptAsymmetric(
+      data,
+      peerPublicKey,
+      encryptionKey.secretKey,
+    );
+    const keyUri = `${didDocument.uri}${getDidEncryptionKey(didDocument).id}`;
```

Comment:
There was an `assembleUri` but since the simple concatenation works now, it was me who suggested we remove it. The code is not particularly elegant, but very straightforward.